osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -359,6 +359,11 @@ void rollback()
 
                         PreImport(item);
 
+                        write.Context.ChangeTracker.Entries()
+                             .Where(e => e.State == EntityState.Unchanged)
+                             .ToList()
+                             .ForEach(e => e.State = EntityState.Detached);
```

Comment:
this "detach everything and try saving" actually seems like a more promising way to do all of this. if we use that approach everywhere, will it work?

just feels like a simpler way to deal.

Reply:
Unfortunately it only solves the problem when entities already in ChangeTracker before calling `Add`, but entities with same id in `item` will still cause tracking conflict issues.